packages/components/pie-cookie-banner/test/unit/localisation-utils.spec.ts

```diff
@@ -0,0 +1,108 @@
+import {
+    describe,
+    it,
+    expect,
+} from 'vitest';
+
+import { localiseText, localiseRichText } from '../../src/localisation-utils';
+
+describe('localiseText', () => {
+    describe('given insuficient parameters it will throw', () => {
+        it('when locale is empty', () => {
+            // @ts-expect-error - Testing invalid input
+            expect(() => localiseText(undefined, 'key')).toThrowError('"locale" parameter cannot be empty');
+        });
+
+        it('when key is empty', () => {
+            // @ts-expect-error - Testing invalid input
+            expect(() => localiseText({}, undefined)).toThrowError('"key" parameter cannot be empty');
+        });
+    });
+
+    describe('given suficient parameters', () => {
+        describe('when the key exists', () => {
+            it('it will return the value', () => {
+                const locale = {
+                    keyA: {
+                        keyA1: 'valueA1',
+                    },
+                };
+
+                // @ts-expect-error - Testing a generic input
+                expect(localiseText(locale, 'keyA.keyA1')).toBe('valueA1');
+            });
+        });
+
+        describe('when the key does not exist', () => {
+            it('it will return the key', () => {
+                const locale = {
+                    keyA: {
+                        keyA1: 'valueA1',
+                    },
+                };
+
+                // @ts-expect-error - Testing a generic input
+                expect(localiseText(locale, 'keyA.keyA2')).toBe('keyA.keyA2');
```

Comment:
From what I can see, the method returns an empty string if the key does not exist?

https://github.com/justeattakeaway/pie/pull/936/files#diff-bec7f7f544dadc684de1dab89d27710e7ef07a6153baca1feb46ad95e993462dR31

Reply:
If the key doesn't exist, both `localiseText` and `localiseRichText` fall back to the key string, making it easy to notice what's missing. An error message is also logged.

Reply:
Added a comment to the functions JSDocs.